File: check_max_chroma.py
#!/usr/bin/env python3
"""Check maximum chroma for green spec"""

# Check maximum chroma for hue=7.877 GY, value=8
# The spec is [7.877, 8.0, 22.595, 4]

from colour.notation.munsell.renotation import \
    maximum_chroma_from_renotation

# Check maximum chroma
hue = 7.877
value = 8.0
code = 4  # GY

max_chroma = maximum_chroma_from_renotation(hue, value, code)
print(f"Maximum chroma for hue={hue:.3f} value={value} code={code}: {max_chroma}")

# Try 7.5GY instead
hue = 7.5
max_chroma = maximum_chroma_from_renotation(hue, value, code)
print(f"Maximum chroma for hue={hue:.3f} value={value} code={code}: {max_chroma}")

# Try value 9
value = 9.0
hue = 7.877
max_chroma = maximum_chroma_from_renotation(hue, value, code)
print(f"Maximum chroma for hue={hue:.3f} value={value} code={code}: {max_chroma}")

File: src/bin/check_max_chroma.rs
//! Check maximum chroma for debugging

use munsellspace::python_port::maximum_chroma_from_renotation;

fn main() {
    // Check maximum chroma for the green spec
    // The spec is [7.877, 8.0, 22.595, 4]
    let hue = 7.877;
    let value = 8.0;
    let code = 4; // GY
    
    let max_chroma = maximum_chroma_from_renotation(hue, value, code);
    println!("Maximum chroma for hue={:.3} value={} code={}: {:.1}", 
             hue, value, code, max_chroma);
    
    // Try 7.5GY instead
    let hue = 7.5;
    let max_chroma = maximum_chroma_from_renotation(hue, value, code);
    println!("Maximum chroma for hue={:.3} value={} code={}: {:.1}", 
             hue, value, code, max_chroma);
    
    // Try value 9
    let value = 9.0;
    let hue = 7.877;
    let max_chroma = maximum_chroma_from_renotation(hue, value, code);
    println!("Maximum chroma for hue={:.3} value={} code={}: {:.1}", 
             hue, value, code, max_chroma);
             
    // Try 7.5GY value 9
    let hue = 7.5;
    let max_chroma = maximum_chroma_from_renotation(hue, value, code);
    println!("Maximum chroma for hue={:.3} value={} code={}: {:.1}", 
             hue, value, code, max_chroma);
}